### PR TITLE
DFPL 1507 Case File View - Correspondence

### PR DESCRIPTION
**Pull request by chak-shing-lo-justice**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFPL-1507

### Change description ###
Update correspondence mapping for case file view

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseDataParent.java [hunk 1]
```diff
@@ -74,6 +74,11 @@ public class CaseDataParent {
     protected final List<Element<ManagedDocument>> previousProceedingListCTSC;
     protected final List<Element<ManagedDocument>> previousProceedingListRemoved;
 
+    protected final List<Element<ManagedDocument>> correspondenceDocList;
+    protected final List<Element<ManagedDocument>> correspondenceDocListLA;
+    protected final List<Element<ManagedDocument>> correspondenceDocListCTSC;
+
+
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseDataParent.java:81]: remove extra line

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java [hunk 2]
```diff
@@ -901,4 +904,48 @@ public void rollbackMigratedSkeletonArgumentList(CaseDetails caseDetails) {
...
+        List<Element<ManagedDocument>> correspondenceDocListLA =
+            Stream.of(caseData.getCorrespondenceDocumentsLA(),
+                    caseData.getCorrespondenceDocumentsSolicitor())
+                .flatMap(Collection::stream)
+                .filter(bundleElement -> bundleElement.getValue().isConfidentialDocument())
+                .map(bundleElement ->
+                    element(bundleElement.getId(),
+                        ManagedDocument.builder().document(bundleElement.getValue().getDocument()).build()))
+                .collect(toList());
+
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:929]: remove an extra line

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java [hunk 3]
```diff
@@ -901,4 +904,48 @@ public void rollbackMigratedSkeletonArgumentList(CaseDetails caseDetails) {
...
+                .flatMap(Collection::stream)
+                .filter(bundleElement -> !bundleElement.getValue().isConfidentialDocument())
+                .map(bundleElement ->
+                    element(bundleElement.getId(),
+                        ManagedDocument.builder().document(bundleElement.getValue().getDocument()).build()))
+                .collect(toList());
+
+        List<Element<ManagedDocument>> correspondenceDocListLA =
+            Stream.of(caseData.getCorrespondenceDocumentsLA(),
+                    caseData.getCorrespondenceDocumentsSolicitor())
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:922]: is there any case that element from`correspondenceDocumentsSolicitor` is uploaded by LA?

Comment (chak-shing-lo-justice) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:995]: Think no, LA solicitor / barrister will go to "manageDocumentsLA" and upload fromcorrespondenceDocumentsLA

Resp / child solicitor goes to "manageDocuments" and upload correspondenceDocumentsSolicitor

Comment (so99y) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:995]: so we don't need `getCorrespondenceDocumentsSolicitor` here?

Comment (chak-shing-lo-justice) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:995]: I think LA level in new Mange Doc flow is for confidential document, right? And CTSC level is for those confidential uploaded by HMCTS staff? 
I am not sure if there is any confidential document uploaded by solicitor or not.

Comment (so99y) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:995]: Yes, `xxxListLA` is for confidential document limited to LA and `xxxListCTSC`is for those confidential uploaded by HMCTS staff.

The confidential checkbox will not be displayed to solicitor. So documents under `correspondenceDocumentsSolicitor` should be non-confidential.

**Review by so99y — APPROVED** [review]
LGTM

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java (hunk 3 above)
Comment (so99y) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:995]: I was just thinking if solicitor uploaded a confidential documents, the documents should be still accessible by them. So , it should be kept under non-confidential collecition.

**Pull request merged**